crates/polars-core/src/series/ops/null.rs

```diff
@@ -51,6 +51,12 @@ impl Series {
                     .collect::<Vec<_>>();
                 StructChunked::new(name, &fields).unwrap().into_series()
             },
+            #[cfg(feature = "object")]
+            DataType::Object(_) => {
+                let series = vec![Series::full_null(name, size, &DataType::Utf8)];
```

Comment:
Why do we return a struct here? This should return an object series.

Reply:
I'm a bit stuck here due to lack of experience.

Ideally, we would be able to do something like:

```
#[cfg(feature = "object")]
DataType::Object(inner_type) => {
    ObjectChunked::full_null(name, size).into_series()
},
```
where (if necessary) `inner_type` could potentially be used to determine/set inner type.

The function full_null is defined as:

```
#[cfg(feature = "object")]
impl<T: PolarsObject> ChunkFullNull for ObjectChunked<T> {
    fn full_null(name: &str, length: usize) -> ObjectChunked<T> {
        let mut ca: Self = (0..length).map(|_| None).collect();
        ca.rename(name);
        ca
    }
}
```
Which if i understand correctly, means that we implement the ChunkFullNull trait on any type that implements the PolarsObject trait.

So in the case of ObjectChunked::full_null(name, size).into_series(), we get the error: `type annotations needed`, because the compiler doesn't know what type T should be. This is because ObjectChunked<T> can be any type that implements the PolarsObject trait, and there could be many types that implement this trait.

What confuses me here is that we want to create an empty series, but because the underlying source data can be of different types, we need to handle this, and this is where I get lost.

Reply:
@ritchie46 push in the right direction here?